hcl2template/parser.go

```diff
@@ -140,16 +136,34 @@ func (p *Parser) Parse(filename string, varFiles []string, argVars map[string]st
 	}
 
 	// Before we go further, we'll check to make sure this version can read
-	// that file, so we can produce a version-related error message rather than
+	// all files, so we can produce a version-related error message rather than
 	// potentially-confusing downstream errors.
 	versionDiags := cfg.CheckCoreVersionRequirements(p.CorePackerVersion)
 	diags = append(diags, versionDiags...)
 	if versionDiags.HasErrors() {
 		return cfg, diags
 	}
 
+	// Decode required_plugins blocks
+	// and create implicit required_plugins blocks
```

Comment:
I think this comment could be expanded. What does it mean to be an implicit required_plugin block? One Packer generates to find HC-maintained plugins for backwards compat?

Reply:
So currently my thinking is that an implicit `required_plugins` happens when you do:

```hcl
source "amazon-ebs" "example" { ... }
```

And no `amazon-ebs` builder can be found. This will then be the equivalent of having

```hcl
packer {
    required_plugins {
        amazon = "latest"
```

( Also note that using `latest` in a normal scenario does not work and packer will ask you to set a version )